Set constructor does not accept Union type of Set | undefined when strictNullChecks is enabled

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨 -->
<!--
Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker.

Please help us by doing the following steps before logging an issue:
  * Search: https://github.com/Microsoft/TypeScript/search?type=Issues
  * Read the CONTRIBUTING guidelines: https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md
  * Read the FAQ: https://github.com/Microsoft/TypeScript/wiki/FAQ
-->

<!-- If you have a QUESTION:
   THIS IS NOT A FORUM FOR QUESTIONS.
   Ask questions at http://stackoverflow.com/questions/tagged/typescript
    or https://gitter.im/Microsoft/TypeScript
-->

<!-- If you have a SUGGESTION:
  Most suggestion reports are duplicates, please search extra hard before logging a new suggestion.
  See https://github.com/Microsoft/TypeScript-wiki/blob/master/Writing-Good-Design-Proposals.md
-->

<!-- If you have a BUG:
  Please fill in the *entire* template below.
-->

<!-- Please try to reproduce the issue with `typescript@next`. It may have already been fixed. -->
**TypeScript Version:**  typescript@3.4.0-dev.20190202

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:** 
set union constructor strict null undefined readonly array strictNullChecks

**Code**

```ts
const params = undefined as Set<any> | undefined;
const ok = params ? new Set(params) : new Set(params);
const ng = new Set(params);
```

compile command
```sh
tsc test.ts -target ES2017 -strict
```

**Expected behavior:**
It should be compiled successfully

**Actual behavior:**
Line `const ng = new Set(params)` raise type error:
```
error TS2345: Argument of type 'Set<any> | undefined' is not assignable to parameter of type 'readonly any[] | null | undefined'.
  Type 'Set<any>' is missing the following properties from type 'readonly any[]': length, concat, join, slice, and 11 more.
```

This error occurs only when strictNullChecks is enabled


**Playground Link:** <!-- A link to a TypeScript Playground "Share" link which demonstrates this behavior --> 
http://www.typescriptlang.org/play/#src=const%20params%20%3D%20undefined%20as%20Set%3Cany%3E%20%7C%20undefined%3B%0D%0Aconst%20ok%20%3D%20params%20%3F%20new%20Set(params)%20%3A%20new%20Set(params)%3B%0D%0Aconst%20ng%20%3D%20new%20Set(params)%3B%0D%0A
Toggle strictNullChecks on/off from "Option" menu

**Related Issues:**
I could not find any related issues.

--- a/src/lib/es2015.iterable.d.ts
+++ b/src/lib/es2015.iterable.d.ts
@@ -177,7 +177,7 @@ interface ReadonlySet<T> {
 }
 
 interface SetConstructor {
-    new <T>(iterable: Iterable<T>): Set<T>;
+    new <T>(iterable?: Iterable<T> | null): Set<T>;
 }
 
 interface WeakSet<T extends object> { }

